modules/features/profile/src/main/java/au/com/shiftyjelly/pocketcasts/profile/ProfileFragment.kt

```diff
@@ -255,19 +261,31 @@ class ProfileFragment : BaseFragment() {
         }
     }
 
-    private fun FragmentProfileBinding.setupEndOfYearPromptCard(isEligible: Boolean) {
-        endOfYearPromptCard.setContent {
-            if (isEligible) {
-                AppTheme(theme.activeTheme) {
-                    EndOfYearPromptCard(
-                        onClick = {
-                            analyticsTracker.track(AnalyticsEvent.END_OF_YEAR_PROFILE_CARD_TAPPED)
-                            // once stories prompt card is tapped, we don't want to show stories launch modal if not already shown
-                            if (settings.getEndOfYearShowModal()) {
-                                settings.setEndOfYearShowModal(false)
-                            }
-                            (activity as? FragmentHostListener)?.showStoriesOrAccount(StoriesSource.PROFILE.value)
-                        },
+    private fun FragmentProfileBinding.setupEndOfYearPromptCard() {
+        bannerCard.setContent {
+            AppTheme(theme.activeTheme) {
+                EndOfYearPromptCard(
+                    onClick = {
+                        analyticsTracker.track(AnalyticsEvent.END_OF_YEAR_PROFILE_CARD_TAPPED)
+                        // once stories prompt card is tapped, we don't want to show stories launch modal if not already shown
+                        if (settings.getEndOfYearShowModal()) {
+                            settings.setEndOfYearShowModal(false)
+                        }
+                        (activity as? FragmentHostListener)?.showStoriesOrAccount(StoriesSource.PROFILE.value)
+                    },
+                )
+            }
+        }
+    }
+
+    private fun FragmentProfileBinding.setupKidsCard() {
+        bannerCard.setContent {
+            AppTheme(theme.activeTheme) {
+                val isKidsBannerVisible by viewModel.isKidsBannerVisible.observeAsState(viewModel.settings.getShowKidsBanner())
```

Comment:
This logic is rather strange. View model should already provide observable state with the initial value. If you migrate `showKidsBanner` setting to `UserSetting` all of this and logic in the biew model can be simplified to something like this.

```kotlin
val showKidsBanner by settings.showKidsBanner.flow.collectAsState() // Or hide it behind the view model
val isKidsBannerVisible = showKidsBannerSetting && FeatureFlag.isEnabled(Feature.KIDS_PROFILE)
```